Work orders create finalized

JorisdeVos opened the pull request:
Work orders can now be made by staff with the role maintenance via their dashboard

JorisdeVos committed:
Merge branch 'main' of https://github.com/MaykeKerssens/BarrocIntens

JorisdeVos committed:
migration and front-end start workorders create

Finally got the frontend and migrations right for the workorder feature, backend still needs to be patched

JorisdeVos committed:
work order create finalized

Work orders can now be created by maintenance workers via their dashboard

MaykeKerssens commented on resources/views/maintenance/workOrder/create.blade.php:
```diff
@@ -0,0 +1,62 @@
+    </x-slot>
+
+    <div class="py-8">
+        <div class="max-w-7xl mx-auto px-4 py-5 bg-white shadow overflow-hidden">
```
Voeg hier error messages toe, zodat de user meer feedback heeft:
           
            @if ($errors->any())
                <div class="bg-red-500 text-white font-bold p-4">
                    <ul>
                        @foreach ($errors->all() as $error)
                            <li>{{ $error }}</li>
                        @endforeach
                    </ul>
                </div>
            @endif

MaykeKerssens commented on resources/views/maintenance/workOrder/create.blade.php:
```diff
@@ -0,0 +1,62 @@
+                                @endforeach
+                            </select>
+                        </div>
+                    <x-primary-button>
```
Button aanpassen, zodat ie niet full-width in neemt

MaykeKerssens commented on app/Http/Controllers/WorkOrderController.php:
```diff
@@ -0,0 +1,45 @@
+        // Attach products to the work order
+        $workOrder->products()->attach($request->input('products'));
+
+        return redirect()->route('workOrder.create')->with('success', 'Work Order created successfully.');
```
Redirect naar Index ipv naar create pagina

MaykeKerssens commented on app/Http/Controllers/WorkOrderController.php:
```diff
@@ -0,0 +1,45 @@
+{
+    public function create()
+    {
+        $products = Product::all(); // Assuming you have a Product model
```
Remove chatGPT comment

MaykeKerssens commented on database/migrations/2024_01_09_082753_create_work_orders_table.php:
```diff
@@ -0,0 +1,31 @@
+            $table->id();
+            $table->string('name');
+            $table->text('description');
+            $table->foreignId('maintenance_appointment_id')->constrained('maintenance_appointments');
```
Ik zou het zo doen, omdat we bij elke andere Migration het zo doen.
            $table->foreignId('maintenance_appointment_id')
                ->references('id')
                ->on('maintenance_appointments');

MaykeKerssens commented on database/migrations/2024_01_09_082847_create_product_work_orders_table.php:
```diff
@@ -0,0 +1,29 @@
+    {
+        Schema::create('product_work_orders', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('product_id');
+            $table->unsignedBigInteger('work_order_id');
```
Ik zou het zo doen, omdat we bij elke andere Migration het zo doen.
        $table->foreignId('product_id')
            ->references('id')
            ->on('products');
        $table->foreignId('work_order__id')
            ->references('id')
            ->on('work_order');

MaykeKerssens commented on resources/views/maintenance/index.blade.php:
```diff
@@ -3,10 +3,27 @@
 
-    <!--Overview for all normal maintenance employees -->
+    <!-- Overview for all normal maintenance employees -->
     <div class="py-8">
         <div class="max-w-7xl mx-auto px-4 py-5 bg-white shadow overflow-hidden">
```
Voeg nog even succes messages toe op deze pagina
`    <div class="max-w-7xl mx-auto my-8 bg-white shadow overflow-hidden">
        @if (session('message'))
            <div class="bg-yellow text-gray-800 font-bold p-4">
                <p>{{ session('message') }}</p>
            </div>
        @endif
        <div class="px-4 py-5">`

JorisdeVos committed:
review changes applied

fixed button width, normalized migration files. succes and error messages placed where needed

JorisdeVos committed:
Deleted not needed invoice product model

this is so i can actually start solving the merge conflict

JorisdeVos committed:
Merge branch 'main' into workOrder-backend-basic-frontend
